src/BlazorComponents/Shared/ChartBase.cs

```diff
@@ -0,0 +1,32 @@
+﻿using BlazorComponents.ChartJS;
+using Microsoft.AspNetCore.Blazor.Components;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace BlazorComponents.Shared
+{
+    public abstract class ChartBase : BlazorComponent
+    {
+
+        [Parameter]
+        protected int Width { get; set; } = 200;
+
+        [Parameter]
+        protected int Height { get; set; } = 200;
+
+        protected abstract IChart GetChart();
```

Comment:
This is some good progress. I just had a question about the usage of `GetChart()`. I guess I don't understand what the usage is? 

Isn't the chart object available already in the needed places?

Reply:
If you have a look at the new "ChartBase" class which is the base class of all charts (cshtml) you will notice the UpdateSize method which needs the Canvas id of the current chart in JS. I do not want to implement this method in every chart type (ChartJsBarChart, ChartJsLineChart, ...). So I created an interface for forcing base properties for accessing in base class. Perhaps it is enough to return the canvas id, not a complete IChart object but I thought maybe they will be more than the canvasId

Reply:
:thumbsup: